doc/rfcs/003-quickly-collaborate-with-coworkers-and-friends.md

```diff
@@ -0,0 +1,147 @@
+    - Filtering your list of past collaborators
+    - Marking certain collaborators as "favorites"
+- Offering to install Teletype if you click on a portal URL and don't yet have Teletype installed ([#220](https://github.com/atom/teletype/issues/220))
+- Selecting a past collaborator and asking to join their portal
+- Selecting multiple past collaborators and inviting them to your portal simultaneously. (In the meantime, you can select a collaborator and invite them, then select another collaborator and invite them, and so on.)
+
+## Drawbacks
+
+- Some people might feel uneasy knowing that past collaborators can see if they are "online" (i.e., if they have Atom open and they're signed into Teletype).
+- Once Teletype provides a list of past collaborators, people may want to be able to chat with those collaborators from within Teletype. While we may eventually want to support chat, any chat-related functionality must exist in service of Teletype's primary vision of "making it as easy to code together as it is to code alone." We'll need to be diligent to avoid scope creep.
+- Depending on the technical solution we choose for implementing presence, we may incur increased server-side resource consumption and/or we may take on additional operational complexity.
+
+## Rationale and alternatives
+
+##### Why is this approach the best in the space of possible approaches?
+
+With the approach described above, we believe Teletype can provide a streamlined set-up process while also preventing harassment. In order for Teletype to prompt you to join someone's portal, you must first have collaborated with that person in the past (by sending your portal URL to them via a third-party service or by joining their portal via a URL they sent you via a third-party service) *and* that person must still exist in your list of past collaborators (i.e., you haven't removed them from your list of past collaborators). In other words, people that you've never collaborated with cannot cause portal invitations to appear in your editor. And people that you've collaborated with in the past, but don't want to receive invitations from in the future, cannot cause portal invitations to appear in your editor either.
+
+##### What other approaches have been considered and what is the rationale for not choosing them?
+
+- Teletype could allow you to enter a person's GitHub username (or their email address) to invite them to your portal. This would remove the need for sharing a URL via a third-party service in order to collaborate, and it would reduce the need to provide a list of past collaborators (i.e., you could just enter a username each time you want to collaborate). However, it would make it possible for any GitHub user to cause invitations to appear inside your Atom instance. This introduces a vector for harassment, so we're avoiding this approach.
```

Comment:
I think this is a really clever way of structuring this to avoid harassment, and it makes complete sense as a place to start. Eventually, I would *still* really like to offer a solution where I can reach any GitHub user via their user name. It seems like giving someone the option to block a single invite or block all invites from first time collaborators might be enough of a recourse in the case of abuse. Then someone would have receive at most *one* unwanted invitation before disabling it. However, for now I think your solution is elegant and anything further would build upon it.

Reply:
It's possible to do this, with some caveats: 

- Teletype needs to respect blocks, spammy status, and suspended status 
- Mutual follows could be a strong indicator of trust
- we should still drop them into the "Join pairing session" flow and ensure explicit consent every time

Getting even one unwanted invitation can be traumatic, so I would prefer we reduce this abuse vector as much as possible 😉